hazelcast-jet-sql/src/test/java/com/hazelcast/jet/sql/SqlLimitTest.java

```diff
@@ -0,0 +1,206 @@
+        );
+
+        assertRowsAnyOrder(
+                "SELECT name FROM " + tableName + " LIMIT 5",
+                asList(new Row("Alice"), new Row("Bob"), new Row("Joey"))
+        );
+    }
+
+    @Test
+    public void negativeLimitValue() {
+        String tableName = createTable(
+                new String[]{"Alice", "1"},
+                new String[]{"Bob", "2"},
+                new String[]{"Joey", "3"}
+        );
+
+        checkFailure0("SELECT name FROM " + tableName + " LIMIT -10", SqlErrorCode.PARSING, "Encountered \"-\"");
+    }
+
+    @Test
+    public void floatNumber_asLimitValue() {
+        String tableName = createTable(
+                new String[]{"Alice", "1"},
+                new String[]{"Bob", "2"},
+                new String[]{"Joey", "3"}
+        );
+
+        assertContainsSubsetOfRows(
+                "SELECT name FROM " + tableName + " LIMIT 2.99",
+                2,
+                asList(new Row("Alice"), new Row("Bob"), new Row("Joey"))
+        );
+
+        assertContainsSubsetOfRows(
+                "SELECT name FROM " + tableName + " LIMIT 2.5",
+                2,
+                asList(new Row("Alice"), new Row("Bob"), new Row("Joey"))
+        );
+
+        assertContainsSubsetOfRows(
+                "SELECT name FROM " + tableName + " LIMIT 2.45",
+                2,
+                asList(new Row("Alice"), new Row("Bob"), new Row("Joey"))
+        );
+
+        assertContainsSubsetOfRows(
+                "SELECT name FROM " + tableName + " LIMIT 3.1",
+                3,
+                asList(new Row("Alice"), new Row("Bob"), new Row("Joey"))
+        );
+    }
+
+    private static String createTable(String[]... values) {
+        String name = randomName();
+        TestBatchSqlConnector.create(
+                sqlService,
+                name,
+                asList("name", "distance"),
+                asList(QueryDataType.VARCHAR, QueryDataType.INT),
+                asList(values)
+        );
+        return name;
+    }
+
+    @Test
+    public void limitOverStream() {
+        assertContainsOnlyOneOfRows(
+                "SELECT * FROM TABLE(GENERATE_STREAM(5)) LIMIT 1",
+                Collections.singletonList(new Row(0L))
+        );
+
+        assertContainsSubsetOfRows(
+                "SELECT * FROM TABLE(GENERATE_STREAM(5)) LIMIT 2",
+                2,
+                asList(new Row(0L),
+                new Row(1L))
+        );
+
+        assertContainsSubsetOfRows(
+                "SELECT * FROM TABLE(GENERATE_STREAM(5)) LIMIT 10",
+                10,
+                asList(new Row(0L),
+                new Row(1L),
+                new Row(2L),
+                new Row(3L),
+                new Row(4L),
+                new Row(5L),
+                new Row(6L),
+                new Row(7L),
+                new Row(8L),
+                new Row(9L))
+        );
+    }
+
+    protected void checkFailure0(
```

Comment:
Could not it be simply replaced with the usage of `Assertions.assertThatThrownBy()` ?